src/log.cpp

```diff
@@ -232,11 +248,11 @@ void Logger::log(LogLevel lev, const std::string &text)
 
 	const std::string thread_name = getThreadName();
 	const std::string label = getLevelLabel(lev);
+	const std::string timestamp = getTimestamp();
 	std::ostringstream os(std::ios_base::binary);
-	os << getTimestamp() << ": " << label << "[" << thread_name << "]: " << text;
+	os << timestamp << ": " << label << "[" << thread_name << "]: " << text;
 
-	logToSystem(lev, text);
-	logToOutputs(lev, os.str());
+	logToOutputs(lev, os.str(), timestamp, thread_name, text);
```

Comment:
This is a good design choice.  We needed the formatting done at a later point in time since some future log outputs might have its own way of associating timestamps or severity levels.